packages/core/src/dialects/postgres/query-generator-typescript.ts

```diff
@@ -77,6 +77,27 @@ export class PostgresQueryGeneratorTypeScript extends AbstractQueryGenerator {
     ]);
   }
 
+  getForeignKeyQuery(tableName: TableNameOrModel, columnName?: string) {
+    if (columnName) {
```

Comment:
we should filter this method by catalog (database) too

Reply:
Actually it filters by default on the current catalog with this implementation because these introspection tables are owned by that catalog, so we can't do that